code/modules/antagonists/traitor/objectives/final_objective/infect_ai.dm

```diff
@@ -0,0 +1,56 @@
+/datum/traitor_objective/ultimate/infect_ai
+	name = "Infect the station AI with an experimental virus."
+	description = "Infect the station AI with an experimental virus. Go to %AREA% to receive an infected law upload board \
+		and use it on the AI core or a law upload console."
+
+	///area type the objective owner must be in to recieve the law upload module
+	var/area/board_area_pickup
+	///checker on whether we have sent the law upload module
+	var/sent_board = FALSE
+
+/datum/traitor_objective/ultimate/infect_ai/can_generate_objective(generating_for, list/possible_duplicates)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	for(var/mob/living/silicon/ai/ai in GLOB.ai_list)
+		if(!ai.stat || ai.mind?.has_antag_datum(/datum/antagonist/malf_ai) || !is_station_level(ai.z))
```

Comment:
Why are you treating `ai.stat` as a boolean? This means that any conscious AI on the station (read: practically all of them) will be skipped over. Is that what you're aiming for, or are you trying to do something like: `if(ai.stat == DEAD)`? `DEAD` is a macro define that resolves to the number `4`, while `CONSCIOUS` is a macro define that resolves to the number `0`, which is falsey in DM.

Change this up, or explain why you're doing it like this. It's very odd.

Reply:
Changed it